dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/telnet/InvokeTelnetHandler.java

```diff
@@ -240,10 +128,96 @@ public String telnet(Channel channel, String message) {
         return buf.toString();
     }
 
+
     private boolean isServiceMatch(String service, ProviderModel provider) {
         return provider.getServiceName().equalsIgnoreCase(service)
                 || provider.getServiceInterfaceClass().getSimpleName().equalsIgnoreCase(service)
                 || provider.getServiceInterfaceClass().getName().equalsIgnoreCase(service)
                 || StringUtils.isEmpty(service);
     }
+
+    private List<Method> findSameSignatureMethod(List<ProviderMethodModel> methods, String lookupMethodName, List<Object> args) {
+        List<Method> sameSignatureMethods = new ArrayList<>();
+        for (ProviderMethodModel model : methods) {
+            Method method = model.getMethod();
+            if (method.getName().equals(lookupMethodName) && method.getParameterTypes().length == args.size()) {
+                sameSignatureMethods.add(method);
+            }
+        }
+        return sameSignatureMethods;
+    }
+
+
+    private Method findInvokeMethod(List<Method> methods, List<Object> args) {
```

Comment:
Looking at the api implementation it seems to be this can become a util method which others can use in future. Would you mind to extract as an util method. Before that would you check whether we can use any of the ClassHelper or ReflectionUtils any method which does the same thing? 
Nice effort of refactoring :)